File: bonsoir_darwin/darwin/Classes/SuccessObject.swift
import Foundation
import Network

/// Sent to the event channel when there is no error.
class SuccessObject {
    /// The response id.
    let id: String

    /// The response service.
    let service: BonsoirService?

    /// Creates a new instance with any result.
    public convenience init(id: String) {
        self.init(id: id, service: nil)
    }

    /// Creates a new success object instance.
    public init(id: String, service: BonsoirService?) {
        self.id = id
        self.service = service
    }

    /// Converts the current instance into a dictionary.
    public func toJson() -> [String: Any] {
        var json: [String: Any] = [:]
        json["id"] = id
        if service != nil {
            json["service"] = service!.toJson()
        }
        return json
    }
}

File: bonsoir_android/android/src/main/kotlin/fr/skyost/bonsoir/SuccessObject.kt
package fr.skyost.bonsoir

import kotlin.collections.HashMap

/**
 * Sent to the event channel when there is no error.
 *
 * @param id The response id.
 * @param service The response service.
 */
data class SuccessObject(private val id: String, private val service: BonsoirService? = null) {
    /**
     * Converts the current instance into a map.
     *
     * @return The map.
     */
    fun toJson(): Map<String, Any> {
        val json: HashMap<String, Any> = HashMap()
        json["id"] = id
        if (service != null) {
            json["service"] = service.toJson()
        }
        return json
    }
}
